homeassistant/components/yolink/__init__.py

```diff
@@ -110,11 +110,20 @@ def on_message_callback(message: tuple[str, BRDP]) -> None:
         device_coordinators[device.device_id] = device_coordinator
     hass.data[DOMAIN][entry.entry_id][ATTR_COORDINATORS] = device_coordinators
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    async def shutdown_subscription(event) -> None:
+        """Shutdown mqtt message subscription."""
+        await yolink_mqtt_client.shutdown_home_subscription()
+
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, shutdown_subscription)
```

Comment:
```suggestion
    entry.async_on_unload(
        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, shutdown_subscription)
    )
```

Reply:
This make sure you don't call the `shutdown_subscription` if the entry was unloaded before shutdown

Reply:
Thanks, changed